src/OpenTelemetry/Metrics/AggregatorStore.cs

```diff
@@ -994,13 +987,9 @@ private void UpdateLongMetricPoint(int metricPointIndex, long value, ReadOnlySpa
         if (metricPointIndex < 0)
         {
             Interlocked.Increment(ref this.DroppedMeasurements);
-
-            if (this.EmitOverflowAttribute)
-            {
-                this.InitializeOverflowTagPointIfNotInitialized();
-                this.metricPoints[1].Update(value);
-            }
-            else if (Interlocked.CompareExchange(ref this.metricCapHitMessageLogged, 1, 0) == 0)
+            this.InitializeOverflowTagPointIfNotInitialized();
+            this.metricPoints[1].Update(value);
+            if (Interlocked.CompareExchange(ref this.metricCapHitMessageLogged, 1, 0) == 0)
```

Comment:
See https://github.com/open-telemetry/opentelemetry-dotnet/pull/5909/files#r1805833875
I think, previously, when this feature was enabled, we were not doing the internal logging. Its best to stick with that behavior. i.e we can remove this log message completely.

Reply:
If someone was relying on listening to this log to detect that limit was hit, then they will no longer be able to do so. However, I still think we don't need this log message anymore. Previous SDK was "dropping" measurements on the floor, so the log made sense. Now SDK never drops a entire measurement, instead replaces given attributes with overflow, so overall metric is accurate. 

Looking for the presence of this attribute is good enough for users to know that limit was hit, so no need to rely on an internal log anymore.

Reply:
I thought it might be useful to keep it for debugging purpose. But I'm ok either way.

Reply:
I've removed the code for internal log emission.